server/controllers/appointments/appointmentsSearch.js

```diff
@@ -0,0 +1,39 @@
+const { getAppointmentsByPatientName } = require('../../database/queries');
+const { appointmentsSearchSchema } = require('../../utils/validation');
+const { boomify } = require('../../utils');
+
+const appointmentsSearch = async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      phone,
+    } = await appointmentsSearchSchema.validate(req.query, {
+      abortEarly: false,
+    });
+
+    if ((firstName && lastName) || phone) {
```

Comment:
I think you don't need to link the first name with last name by && operator, it's good to let the route work by supplying any of these three values.